### PR TITLE
AI fix for big unit, enable AI UTs after refactoring

### PR DESCRIPTION
**Pull request by mFla**
Changes in the PR

- enable some UTs for AI after code refactoring
- fix bug when AI with big unit could attack though one cell

### CODE REVIEW

File: game/core/package.json [hunk 1]
```diff
@@ -72,7 +72,8 @@
         "uuid": "^10.0.0"
     },
     "devDependencies": {
-        "@parcel/config-default": "^2.12.0",
```
Comment (zolotukhin) [inline game/core/package.json:75]: why do we remove that?

File: game/core/package.json [hunk 2]
```diff
@@ -72,7 +72,8 @@
...
+        "@babel/preset-typescript": "^7.24.7",
+        "globals": "^15.9.0",
```
Comment (zolotukhin) [inline game/core/package.json:76]: What is it needed for? Also yarn.lock has to be updated, I do not see it has been committed

File: game/core/package.json [hunk 3]
```diff
@@ -72,7 +72,8 @@
         "uuid": "^10.0.0"
     },
     "devDependencies": {
-        "@parcel/config-default": "^2.12.0",
+        "@babel/preset-typescript": "^7.24.7",
```
Comment (zolotukhin) [inline game/core/package.json:75]: Does it still give a warning on build?

File: game/core/package.json (hunk 1 above)
Comment (mFla) [reply game/core/package.json]: messed up with the changes for the file, thanks for pointing!

File: game/core/package.json (hunk 2 above)
Comment (mFla) [reply game/core/package.json]: messed up with the changes for the file, thanks for pointing!

File: game/core/package.json (hunk 3 above)
Comment (mFla) [reply game/core/package.json]: messed up with the changes for the file, thanks for pointing!

**Pull request merged**